serverless-functions/src/functions/features/callback-and-voicemail/studio/wait-experience.protected.js

```diff
@@ -231,12 +246,83 @@ exports.handler = async (context, event, callback) => {
       );
       return callback(null, twiml);
 
+    case 'handle-callback-choice':
+      if (Digits && Digits === '1') {
+        twiml.redirect(
+          `${domain}/features/callback-and-voicemail/studio/wait-experience?mode=submit-callback&CallSid=${CallSid}&enqueuedTaskSid=${enqueuedTaskSid}&to=${event.Caller}`,
+        );
+        return callback(null, twiml);
+      } else if (Digits && Digits === '2') {
+        // Get phone number from customer as input to request a callback
+        const gather = twiml.gather({
+          input: 'dtmf',
+          timeout: 10,
+          numDigits: 13,
+          finishOnKey: '#',
+          action: `${domain}/features/callback-and-voicemail/studio/wait-experience?mode=handle-callback-for-other-number-confirmation-option&enqueuedTaskSid=${enqueuedTaskSid}&CallSid=${CallSid}`,
+          method: 'GET',
+        });
+        gather.say(options.sayOptions, options.messages.callbackForOtherNumber);
+      }
+
+      // Loop back to the start of the wait loop
+      twiml.redirect(
+        `${domain}/features/callback-and-voicemail/studio/wait-experience?mode=main-wait-loop&CallSid=${CallSid}&enqueuedTaskSid=${enqueuedTaskSid}&skipGreeting=true`,
+      );
+      return callback(null, twiml);
+
+    case 'handle-callback-for-other-number-confirmation-option':
+      if (Digits) {
+        const say = twiml.say(`You entered `);
+        say.sayAs(
+          {
+            'interpret-as': 'telephone',
+          },
+          Digits.trim(),
+        );
+
+        const gather = twiml.gather({
+          input: 'dtmf',
+          timeout: 15,
+          numDigits: 1,
+          finishOnKey: '#',
+          action: `${domain}/features/callback-and-voicemail/studio/wait-experience?mode=handle-callback-for-other-number-confirmation&enqueuedTaskSid=${enqueuedTaskSid}&updatedPhoneNumber=${Digits.trim()}`,
+          method: 'GET',
+        });
+        gather.say(` press 1 to confirm and 2 to re-enter`);
```

Comment:
Can the message string be extracted into the `options.messages` object to allow easier customizing/localizing?